### PR TITLE
Update mapfish print to most recent updates

### PR DESCRIPTION
**Pull request by offtherailz**
## Description

This PR closes https://github.com/geosolutions-it/mapfish-print/issues/65

We need to remove the pom from our repository to avoid any possible conflict between downloads (it may be randomly downloaded from one or the other repository, depending on which is the most recent).

We have to decide how to name and possibly how to get the lib from osgeo repo, where possible (this is why it is in draft).

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [x] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

mapfish-print/issues/65

**What is the new behavior?**

Updated to latest gs version.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by offtherailz** [comment]
> We need to remove the pom from our repository to avoid any possible conflict between downloads (it may be randomly downloaded from one or the other repository, depending on which is the most recent).

@tdipisa could you ask devOps to remove the 2.3-SNAPSHOT from our repository?

**Comment by offtherailz** [comment]
tested locally, it works. Ready for test. Please review and merge at your convenience @tdipisa

File: docs/developer-guide/mapstore-migration-guide.md [hunk 1]
```diff
@@ -22,6 +22,38 @@ This is a list of things to check if you want to update from a previous version
 
 ## Migration from 2023.02.xx to 2024.01.00
 
+### MapFish Print update
+
+MapFish Print library has been updated to be aligned to the one used by GeoServer and be able to build with Java 11. (see this issue <https://github.com/geosolutions-it/mapfish-print/issues/65>)
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:27]:
```suggestion
**MapFish Print** library has been updated to be aligned to the one used by GeoServer and be able to build with Java 11. (see this issue <https://github.com/geosolutions-it/mapfish-print/issues/65>)
```

File: docs/developer-guide/mapstore-migration-guide.md [hunk 2]
```diff
@@ -22,6 +22,38 @@ This is a list of things to check if you want to update from a previous version
...
+For this reason, if you are using mapfish-print, you may have to update your project
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:28]:
```suggestion
For this reason, if you are using mapfish-print, you have to update your project
```

File: docs/developer-guide/mapstore-migration-guide.md [hunk 3]
```diff
@@ -22,6 +22,38 @@ This is a list of things to check if you want to update from a previous version
...
+                <!-- mapfish-print -->
+                <dependency>
+                    <groupId>org.mapfish.print</groupId>
+                    <artifactId>print-lib</artifactId>
+-                    <version>geosolutions-2.3-SNAPSHOT</version>
++                    <version>2.3-SNAPSHOT</version>
+
+```
+
+And adding the repository where this library is hosted in the `repositories` section of the same `pom.xml`
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:42]:
```suggestion
And adding the repository where this library is hosted in the `repositories` section of the same `pom.xml` (usually in `web` folder of a project)
```

**Comment by offtherailz** [comment]
note: I applied a few improvements to the doc by myself a few seconds after asking the review.

**Review by tdipisa — APPROVED** [review]

**Comment by tdipisa** [comment]
@offtherailz first problem using _npm start_  the backend doesn't start properly now

**Pull request merged**